pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -611,11 +611,18 @@ func getInstallerPodName(revision int32, nodeName string) string {
 }
 
 // ensureInstallerPod creates the installer pod with the secrets required to if it does not exist already
-func (c *InstallerController) ensureInstallerPod(nodeName string, operatorSpec *operatorv1.StaticPodOperatorSpec, revision int32) error {
+func (c *InstallerController) ensureInstallerPod(nodeName string, operatorSpec *operatorv1.StaticPodOperatorSpec, revision int32, revisionReason string) error {
 	pod := resourceread.ReadPodV1OrDie(bindata.MustAsset(filepath.Join(manifestDir, manifestInstallerPodPath)))
 
 	pod.Namespace = c.targetNamespace
 	pod.Name = getInstallerPodName(revision, nodeName)
+	if pod.Annotations == nil {
+		pod.Annotations = map[string]string{}
+	}
+
+	// Report the reason this installer pod was created.
+	pod.Annotations["operator.openshift.io/revision-reason"] = revisionReason
```

Comment:
operator.openshift.io is pretty generic. staticpod.operator.openshift.io maybe?

Reply:
sure